extensions/amp-instagram/0.1/amp-instagram.js

```diff
@@ -141,6 +151,30 @@ class AmpInstagram extends AMP.BaseElement {
     });
   }
 
+  /** @private */
+  handleInstagramMessages_(event) {
+    if (event.origin != 'https://www.instagram.com' ||
+        event.source != this.iframe_.contentWindow) {
+      return;
+    }
+    if (!event.data ||
+        !(isObject(event.data) || event.data.indexOf('{') == 0)) {
+      return;  // Doesn't look like JSON.
+    }
+    const data = isObject(event.data) ? event.data : tryParseJson(event.data);
+    if (data === undefined) {
+      return; // We only process valid JSON.
+    }
+    if (data.type == 'MEASURE') {
+      const height = data.details.height;
+      if (this.iframe_./*OK*/offsetHeight !== height) {
```

Comment:
Makes sense to be in vsync. `attemptChangeHeight` does its own mutate so I think we can do the check and the `attemptChangeHeight` in the same measure block. e.g.: 
```
    const height = data.details.height;
    this.getVsync().measure(() => {
      if (this.iframe_./*OK*/offsetHeight !== height) {
         this.attemptChangeHeight(height - PADDING_TOP).catch(() => {});
      }
    });
```